docs/configuring_previous_line_rules.rst

```diff
@@ -0,0 +1,133 @@
+Configuring Previous Line Rules
+-------------------------------
+
+There are rules which will check the contents on lines above code structures.
+These rules allow enforcement of comments and blank lines.
+
+There are several options to these rules, which can be selected by using the :code:`method` option:
+
++---------------------+--------------------------------------------------+
+| Method              | Description                                      |
++=====================+==================================================+
+| require_blank_line  | Requires a blank line on the line above.         |
++---------------------+--------------------------------------------------+
+| no_code             | Either a blank or comment on the line above.     |
++---------------------+--------------------------------------------------+
+| allow_comment       | Either a blank or comment on line above, with a  |
+|                     | blank line above the comment.                    |
++---------------------+--------------------------------------------------+
+| require_comment     | Requires a comment on the line above, with a     |
+|                     | blank line above the comment.                    |
```

Comment:
Sorry, first time giving multi line suggestions on github - I think I missed the header and last line ...
```suggestion
| require_blank_line  | Requires a blank line on the line above.                   |
+---------------------+------------------------------------------------------------+
| no_code             | Either a blank line; or comment(s) on the line(s) above.   |
+---------------------+------------------------------------------------------------+
| allow_comment       | Either a blank line; or comment(s) on line(s) above and a  |
|                     | blank line above the comment(s).                           |
+---------------------+------------------------------------------------------------+
| require_comment     | Comment(s) required on the line(s) above and a             |
|                     | blank line above the comment(s).                           |
```